### PR TITLE
rework on the switch statement for decomposition methods in ComputeFiniteStrain

### LINKED ISSUE #7886: rework on using scoped enum for decomposition methods in ComputeFinteStrain
### Description of the enhancement or error report
use scoped enum to make the switch statement for the decomposition method more efficient

### PR DESCRIPTION
**Pull request by hchen139**
closes #7886

### CODE REVIEW

**Comment by hchen139** [comment]
@bwspenc @permcody Scoped enum for the switch statement in ComputeFiniteStrain.

**Review by dschwen — APPROVED** [review]
:+1: looks nice!

**Comment by bwspenc** [comment]
Looks good.  My only comment is that I don't think you have to use all caps in the input file (because it's not case sensitive), so you can still use the old form:
`decomposition_method = EigenSolution`
which is maybe more consistent with other input parameters.

File: modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C [hunk 1]
```diff
@@ -220,6 +220,6 @@ ComputeFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment,
     }
 
     default:
-    mooseError("ComputeFiniteStrain Error: Pass valid decomposition type: TAYLOREXPANSION or EIGENSOLUTION.");
+    mooseError("ComputeFiniteStrain Error: Pass valid decomposition type: TaylorExpansion or EigenSolution.");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C]: indentation :-D

**Pull request merged**